dev/io.openliberty.security.jakartasec.3.0.internal_fat/fat/src/io/openliberty/security/jakartasec/fat/tests/TokenValidationTests.java

```diff
@@ -0,0 +1,1219 @@
+ * does so correctly.  THese tests will make sure that the token returned from the token endpoint will be validated properly.
+ * We'll use a test token endpoint and have it return tokens created by the test cases - these tokens will include claims with
+ * invalid values, or missing claims.
+ *
+ */
+
+/**
+ * Tests appSecurity-5.0
+ */
+@SuppressWarnings("restriction")
+@RunWith(FATRunner.class)
+public class TokenValidationTests extends CommonAnnotatedSecurityTests {
+
+    protected static Class<?> thisClass = TokenValidationTests.class;
+
+    protected static ShrinkWrapHelpers swh = null;
+
+    @Server("jakartasec-3.0_fat.op")
+    public static LibertyServer opServer;
+    @Server("jakartasec-3.0_fat.tokenValidation.jwt.rp")
+    public static LibertyServer rpJwtServer;
+    @Server("jakartasec-3.0_fat.tokenValidation.opaque.rp")
+    public static LibertyServer rpOpaqueServer;
+
+    public static LibertyServer rpServer;
+
+    @ClassRule
+    public static RepeatTests repeat = createTokenTypeRepeats();
+
+    public static final SecurityFatHttpUtils secHttpUtils = new SecurityFatHttpUtils();
+    public static final JwtTokenBuilderUtils tokenBuilderHelpers = new JwtTokenBuilderUtils();
+
+    int Default_expires_in = 7199;
+    int Omit_expires_in = 9999;
+    String Default_token_type = "Bearer";
+    String Default_scope = "openid profile";
+    String Default_refresh_token = "21MhoIC95diaQo9tb5UpFBDFlHh45NixhcKkCwRipszH6WIzKz";
+
+    Expectations tokenResponseExpectations = null;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+
+        // write property that is used to configure the OP to generate JWT or Opaque tokens
+        rpServer = setTokenTypeInBootstrap(opServer, rpJwtServer, rpOpaqueServer);
+
+        // Add servers to server trackers that will be used to clean servers up and prevent servers
+        // from being restored at the end of each test (so far, the tests are not reconfiguring the servers)
+        updateTrackers(opServer, rpServer, false);
+
+        List<String> waitForMsgs = null;
+        opServer.startServerUsingExpandedConfiguration("server_orig.xml", waitForMsgs);
+        SecurityFatHttpUtils.saveServerPorts(opServer, Constants.BVT_SERVER_1_PORT_NAME_ROOT);
+        opHttpBase = "http://localhost:" + opServer.getBvtPort();
+        opHttpsBase = "https://localhost:" + opServer.getBvtSecurePort();
+
+        rpServer.startServerUsingExpandedConfiguration("server_tokenValidation.xml", waitForMsgs);
+        SecurityFatHttpUtils.saveServerPorts(rpServer, Constants.BVT_SERVER_2_PORT_NAME_ROOT);
+
+        rpHttpBase = "http://localhost:" + rpServer.getBvtPort();
+        rpHttpsBase = "https://localhost:" + rpServer.getBvtSecurePort();
+
+        opServer.addIgnoredErrors(Arrays.asList(MessageConstants.CWWKS1616E_USERINFO_REQUEST_MISSING_ACCESS_TOKEN, MessageConstants.CWWKS1617E_USERINFO_REQUEST_BAD_TOKEN));
```

Comment:
They're added here to avoid multiple instances of the same messages being added multiple times.  Multiple tests encounter these messages and the server method doesn't have the smarts to only add them once...

Reply:
1616 is not only being issued for one test, so, I moved that one to the specific test.
1617 is issued in well over half of the tests - keeping that in setup so we don't end up with 15 - 20 instances of the same message in the list to check.